p24/tests/unit.rs

```diff
@@ -0,0 +1,120 @@
+#[test]
+fn test_f1() {
```

Comment:
These are integration tests. The slides asked to implement unit tests. In other words, you should have these tests in the `refs.rs` file wrapped in:
```rust
#[cfg(test)]
mod tests {
    // ...
}
```